assets/src/editor/blocks/video-promo-container/edit.js

```diff
@@ -0,0 +1,202 @@
+/**
+ * External dependencies
+ */
+import clsx from 'clsx';
+
+/**
+ * WordPress dependencies
+ */
+import { isBlobURL } from '@wordpress/blob';
+import { Spinner } from '@wordpress/components';
+import {
+	useBlockProps,
+	useInnerBlocksProps,
+	store as blockEditorStore,
+} from '@wordpress/block-editor';
+import { __ } from '@wordpress/i18n';
+import { useSelect } from '@wordpress/data';
+
+/**
+ * Internal dependencies
+ */
+import VideoPromoInspectorControls from './inspector-controls';
+import VideoPromoBlockControls from './block-controls';
+import { useMediaSrc } from './shared';
+import { useEffect } from 'react';
+
+function getInnerBlocksTemplate( attributes ) {
+	return [
+		[
+			'core/paragraph',
+			{
+				align: 'center',
+				placeholder: __( 'Write title…' ),
+				...attributes,
+			},
+		],
+	];
+}
+
+/**
+ * Is the URL a temporary blob URL? A blob URL is one that is used temporarily while
+ * the media (image or video) is being uploaded and will not have an id allocated yet.
+ *
+ * @param {number} id  The id of the media.
+ * @param {string} url The url of the media.
+ *
+ * @return {boolean} Is the URL a Blob URL.
+ */
+const isTemporaryMedia = ( id, url ) => ! id && isBlobURL( url );
+
+function VideoPromoEdit( {
+	attributes,
+	clientId,
+	setAttributes,
+} ) {
+	const {
+		videoId,
+		mobileVideoId,
+		allowedBlocks,
+		templateLock,
+		tagName: TagName = 'div',
+		posterId,
+		mobilePosterId,
+		videoUrl,
+		mobileVideoUrl,
+		poster,
+		mobilePoster,
+	} = attributes;
+
+	// Fetch the attachment sources.
+	const _videoUrl = useMediaSrc( videoId ) ?? '';
+
+	const _mobileVideoUrl = useMediaSrc( mobileVideoId ) ?? '';
+	const _poster = useMediaSrc( posterId ) ?? '';
+	const _mobilePoster = useMediaSrc( mobilePosterId ) ?? '';
+
+	// Update URL attributes if they don't match the freshly fetched attachment sources.
+	useEffect(() => {
```

Comment:
It's not a huge performance impact, but technically all this will run every time a single one of these changes. Have you thought about using a separate effect for each?

e.g.

```js

useEffect( () => {

    if ( videoId && _videoUrl && ( _videoUrl !== videoUrl ) ) {
			setAttributes({ videoUrl: _videoUrl });
		}

}, [ __videoUrl, videoId, videoURl ] );
```

Reply:
Done